cli/cfncluster/cfncluster.py

```diff
@@ -206,189 +210,127 @@ def stop(args):
     config = cfnconfig.CfnClusterConfig(args)
 
     # Set Resource limits
-    asg = get_asg(stack_name=stack_name, config=config)
-    set_asg_limits(asg=asg, min=0, max=0, desired=0)
+    asg_name = get_asg_name(stack_name=stack_name, config=config)
+    set_asg_limits(asg_name=asg_name, config=config, min=0, max=0, desired=0)
 
 def list(args):
     config = cfnconfig.CfnClusterConfig(args)
-    cfnconn = boto.cloudformation.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                    aws_secret_access_key=config.aws_secret_access_key)
+    cfn = boto3.client('cloudformation', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
     try:
-        stacks = cfnconn.describe_stacks()
+        stacks = cfn.describe_stacks().get('Stacks')
         for stack in stacks:
-            if stack.stack_name.startswith('cfncluster-'):
-                logger.info('%s' % (stack.stack_name[11:]))
-    except boto.exception.BotoServerError as e:
-        if e.message.endswith("does not exist"):
-            logger.critical(e.message)
-        else:
-            raise e
+            if stack.get('StackName').startswith('cfncluster-'):
+                logger.info('%s' % (stack.get('StackName')[11:]))
     except KeyboardInterrupt:
         logger.info('Exiting...')
         sys.exit(0)
 
 def get_master_server_id(stack_name, config):
     # returns the physical id of the master server
     # if no master server returns []
-    cfnconn = boto.cloudformation.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                    aws_secret_access_key=config.aws_secret_access_key)
-    temp_resources = []
+    cfn = boto3.client('cloudformation', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
 
-    while True:
-        try:
-            resources = cfnconn.describe_stack_resources(stack_name)
-        except boto.exception.BotoServerError as e:
-            if e.message.endswith("does not exist"):
-                logger.critical(e.message)
-                sys.stdout.flush()
-                sys.exit(0)
-            else:
-                raise e
-        temp_resources.extend(resources)
-        if not resources.next_token:
-            break
-        resources = cfnconn.describe_stack_resources(stack, next_token=resources.next_token)
+    try:
+        resources = cfn.describe_stack_resource(StackName=stack_name, LogicalResourceId='MasterServer')
+        return resources.get('StackResourceDetail').get('PhysicalResourceId')
+    except ClientError as e:
+        logger.critical(e.response.get('Error').get('Message'))
+        sys.exit(1)
```

Comment:
In case the error message ends with "does not exist" we are changing the exit code from 0 to 1